src/testdrive/src/action.rs

```diff
@@ -401,6 +405,16 @@ pub async fn build(cmds: Vec<PosCommand>, state: &State) -> Result<Vec<PosAction
         vars.insert(format!("env.{}", key), value);
     }
 
+    for arg in &state.arg_vars {
+        let mut parts = arg.splitn(2, '=');
+        let name = parts.next().expect("Clap ensures all --vars get a value");
+        validate_ident(name).map_err(Error::message)?;
+        let val = parts
+            .next()
+            .ok_or_else(|| Error::message(format!("No =VALUE for --var {}", name)))?;
+        vars.insert(format!("arg.{}", name), val.to_string());
+    }
```

Comment:
Oh, hrm, the point was to keep the argument parsing (knowledge of `--var`, and the specific `=` separator) in main.rs, and to keep the ident validation in lib.rs. It's muddied together like this. It's not worth another CI cycle but I'd like to fix that up at some point.

Reply:
ohhhh, okay, yeah that's much better than this. I've got to go through another CI cycle for merge conflicts anyway, I'll fix it.